jujugui/static/gui/src/app/components/deployment/choose-cloud/test-choose-cloud.js

```diff
@@ -32,20 +32,25 @@ describe('DeploymentChooseCloud', function() {
   });
 
   beforeEach(function() {
-    // XXX huwshimi 22 March 2016: this can be removed once the credentials are
-    // passed in as a prop.
-    credentials = juju.components.DeploymentChooseCloud.prototype.CREDENTIALS;
+    jem = {
+      listTemplates: (callback) => {
+        callback(null, [{path: 'test-owner/test'}]);
+      }
+    };
   });
 
   afterEach(function() {
-    juju.components.DeploymentChooseCloud.prototype.CREDENTIALS = credentials;
+    jem = null;
   });
 
   it('can render without credentials', function() {
-    juju.components.DeploymentChooseCloud.prototype.CREDENTIALS = [];
+    jem.listTemplates = (callback) => {
+      callback(null, []);
```

Comment:
Can you verify that if there are no templates in jem it returns []? I seem to recall that it returned null when we were testing.